src/wmtk/MultiMeshManager.hpp

```diff
@@ -25,49 +31,136 @@ class MultiMeshManager
+    //
+    void register_child_mesh(
+        Mesh& my_mesh,
+        const std::shared_ptr<Mesh>& child_mesh,
+        const std::vector<std::array<Tuple, 2>>& child_tuple_my_tuple_map);
+
+
+    // bool are_maps_valid(const Mesh& my_mesh) const;
+
+    //===========
+    //===========
+    // Map functions
+    //===========
+    //===========
+    // Note that when we map a M-tuplefrom a K-complex to a J-complex there are different
+    // relationships necessary if K == J
+    //    if M == K then this is unique
+    //    if M < K then this is many to many
+    // if K < J
+    //    if M == K then it is one to many
+    //    if M < K then it is many to many
+
+    //===========
+    // Simplex maps
+    //===========
+    // generic mapping function that maps a tuple from "this" mesh to the other mesh
+    std::vector<Simplex> map(const Mesh& my_mesh, const Mesh& other_mesh, const Simplex& my_simplex)
+        const;
+    // generic mapping function that maps a tuple from "this" mesh to the other mesh
+    std::vector<Tuple>
+    map_tuples(const Mesh& my_mesh, const Mesh& other_mesh, const Simplex& my_simplex) const;
 
-    // TODO: make it a free function? static function?
-    // Map source_tuple from source_mesh to target_mesh
-    static Tuple map_tuple_between_meshes(const Mesh& source_mesh, const Mesh& target_mesh, MeshAttributeHandle<long> source_map_handle, const Tuple& source_tuple);
 
-    static std::vector<Simplex> find_all_simplices_in_child_mesh(const Mesh& parent_mesh, long child_id, const Simplex& simplex_parent);
+    Simplex map_to_parent(const Mesh& my_mesh, const Simplex& my_simplex) const;
+    Tuple map_to_parent_tuple(const Mesh& my_mesh, const Simplex& my_simplex) const;
+
+
+    // generic mapping function that maps a tuple from "this" mesh to one of its children
+    std::vector<Simplex>
+    map_to_child(const Mesh& my_mesh, const Mesh& child_mesh, const Simplex& my_simplex) const;
+    std::vector<Tuple> map_to_child_tuples(
+        const Mesh& my_mesh,
+        const Mesh& child_mesh,
+        const Simplex& my_simplex) const;
 
 
     // Utility function to map a edge tuple to all its children, used in operations
-    static std::vector<Tuple> map_edge_tuple_to_all_children(const Mesh& parent_mesh,const Tuple& edge_tuple);
+    std::vector<Tuple> map_edge_tuple_to_all_children(const Mesh& my_mesh, const Simplex& tuple)
+        const;
+
+    const Mesh& get_root_mesh(const Mesh& my_mesh) const;
+    Mesh& get_root_mesh(Mesh& my_mesh);
+
+protected:
+    struct ChildData
+    {
+        std::shared_ptr<Mesh> mesh;
+        // store the map from the manager's mesh to the child mesh (on the top
+        // level simplex of the mesh)
+        // encoded by a pair of two tuples, from a tuple in current mesh to a tuple in
+        // child_mesh
+        MeshAttributeHandle<long> map_handle;
+    };
+
+private:
+    Mesh* m_parent = nullptr;
```

Comment:
it should be a std::weak_ptr i guess - but doing this would require making every `Mesh` a shared_ptr (and also disabling public construction of meshes)